mitodl/micromasters
# Fix race condition resulting in multiple emails sent

#### What are the relevant tickets?
Fixes #3036

#### What's this PR do?
Uses select_for_update to lock the `SentAutomaticEmail` row until after the email was sent. It also refactors `mark_emails_as_sent` to be a context manager used wherever we need to mark automatic emails as sent

#### How should this be manually tested?
Add a few `Education` and `Employment` entries to your profile and save. As a separate user create an `AutomaticEmail` for your user using the learner search page interface. Switch back to your original user, update your profile and you should see only one email sent to that user's email.
